Serializer/Serializer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Serializer;
+
+/**
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+interface Serializer
+{
+    /**
+     * @param mixed  $data
+     * @param string $format
+     * @param mixed  $context
```

Comment:
As @xabbuh said [here](https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1171#issuecomment-158617224) IMO we should the context adapters behavior in the serializer adapters and change this type hint to ``ContextInterface``.
May be better to do this later in another PR as this one becomes really big.